### ISSUE TITLE
ConnectCallback not currently supported

### ISSUE THREAD

**Issue by gibbed** [TalAloni/StandardSocketsHttpHandler#3]
It's not explicitly mentioned in the documentation that `ConnectCallback` is not supported.

https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.connectcallback

My goal was to hopefully use this to force IPv4 connections on pre-.NET 5, as described in this issue comment:

https://github.com/dotnet/runtime/issues/44686#issuecomment-733797994

I don't see a way to replicate this callback with StandardSocketsHttpHandler as it stands currently? Or am I mistaken?

If you don't intend to implement this, that's fine, just the documentation would need updating in that case.

**Comment by TalAloni** [comment]
Hi,
Given that .NET 5 was released AFTER this library was published, the library API is slightly different than the .NET 5 API,
(it might be a good idea to align them when I'll have the time)
For now, you will need to use the SocketCreated event (the library "equivalent" of ConnectCallback)

**Comment by gibbed** [comment]
Unless I'm mistaken, `SocketCreated` wouldn't allow you control over the creation of the socket?

**Comment by TalAloni** [comment]
You are right. Can you please provide a code sample on how you intend to create the socket so I'll able to test against when updating the implementation?

**Comment by gibbed** [comment]
The linked comment in the OP contains the code I'm intending to use, which is used to force creation of an IPv4 socket.

https://github.com/dotnet/runtime/issues/44686#issuecomment-733797994

I'll copy it here for posterity:

```cs
public HttpClient CreateWorkaroundClient()
{
    SocketsHttpHandler handler = new SocketsHttpHandler
    {
        ConnectCallback = IPv4ConnectAsync
    };
    return new HttpClient(handler);

    static async ValueTask<Stream> IPv4ConnectAsync(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
    {
        // By default, we create dual-mode sockets:
        // Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp);

        Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
        socket.NoDelay = true;

        try
        {
            await socket.ConnectAsync(context.DnsEndPoint, cancellationToken).ConfigureAwait(false);
            return new NetworkStream(socket, ownsSocket: true);
        }
        catch
        {
            socket.Dispose();
            throw;
        }
    }
}
```

**Comment by TalAloni** [comment]
OK, I'll try to get to it in the next 48 hours.

**Comment by gibbed** [comment]
Much appreciated! I hadn't realized this library predates .NET 5 as well.

**Comment by TalAloni** [comment]
I went ahead and published v2.2.0.2 which includes ConnectCallback

**Comment by gibbed** [comment]
That was far quicker than I had expected. Thanks for the update!
